contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -404,6 +349,167 @@ contract StakingVault is IStakingVault, OwnableUpgradeable {
         emit Reported(_valuation, _inOutDelta, _locked);
     }
 
+    /**
+     * @notice Returns the 0x02-type withdrawal credentials for the validators deposited from this `StakingVault`
+     *         All consensus layer rewards are sent to this contract. Only 0x02-type withdrawal credentials are supported
+     * @return Withdrawal credentials as bytes32
+     */
+    function withdrawalCredentials() public view returns (bytes32) {
+        return bytes32(WC_0X02_PREFIX | uint160(address(this)));
+    }
+
+    /**
+     * @notice Returns whether deposits are paused by the vault owner
+     * @return True if deposits are paused
+     */
+    function beaconChainDepositsPaused() external view returns (bool) {
+        return _getStorage().beaconChainDepositsPaused;
+    }
+
+    /**
+     * @notice Pauses deposits to beacon chain
+     * @dev    Can only be called by the vault owner
+     */
+    function pauseBeaconChainDeposits() external onlyOwner {
+        ERC7201Storage storage $ = _getStorage();
+        if ($.beaconChainDepositsPaused) {
+            revert BeaconChainDepositsResumeExpected();
+        }
+
+        $.beaconChainDepositsPaused = true;
+
+        emit BeaconChainDepositsPaused();
+    }
+
+    /**
+     * @notice Resumes deposits to beacon chain
+     * @dev    Can only be called by the vault owner
+     */
+    function resumeBeaconChainDeposits() external onlyOwner {
+        ERC7201Storage storage $ = _getStorage();
+        if (!$.beaconChainDepositsPaused) {
+            revert BeaconChainDepositsPauseExpected();
+        }
+
+        $.beaconChainDepositsPaused = false;
+
+        emit BeaconChainDepositsResumed();
+    }
+
+    /**
+     * @notice Performs a deposit to the beacon chain deposit contract
+     * @param _deposits Array of deposit structs
+     * @dev    Includes a check to ensure `StakingVault` is balanced before making deposits
+     */
+    function depositToBeaconChain(Deposit[] calldata _deposits) external {
+        if (_deposits.length == 0) revert ZeroArgument("_deposits");
+
+        ERC7201Storage storage $ = _getStorage();
+
+        if (msg.sender != $.nodeOperator) revert NotAuthorized("depositToBeaconChain", msg.sender);
+        if ($.beaconChainDepositsPaused) revert BeaconChainDepositsArePaused();
+        if (valuation() < $.locked) revert ValuationBelowLockedAmount();
+
+        uint256 totalAmount = 0;
+        uint256 numberOfDeposits = _deposits.length;
+        for (uint256 i = 0; i < numberOfDeposits; i++) {
+            IStakingVault.Deposit calldata deposit = _deposits[i];
+            DEPOSIT_CONTRACT.deposit{value: deposit.amount}(
+                deposit.pubkey,
+                bytes.concat(withdrawalCredentials()),
```

Comment:
wdim?

Reply:
This is hacky way to convert `bytes32` to `bytes memory`, but will replace with `abi.encodePacked()`

Reply:
No need. It's official "hacky" way. encodePacked is mildly deprecated, as far as I know.